example-plugins/example_patient_sync/protocols/patient_sync.py

```diff
@@ -0,0 +1,155 @@
+
+        # Great, now we know if the patient is assigned a system external ID with the partner
+        # platform, and if we need to update it. At this point the system_patient_id can be 3 values:
+        # 1. value we already had stored in Canvas,
+        # 2. value we just got from our GET API lookup, or
+        # 3. None
+        # And we have a true/false call to action: `update_patient_external_identifier`
+
+        # Generate the payload for creating or updating the patient in partner platform API
+        partner_payload = {
+            "externalId": canvas_patient.id,
+            "firstName": canvas_patient.first_name,
+            "lastName": canvas_patient.last_name,
+            "dateOfBirth": canvas_patient.birth_date.isoformat(),
+        }
+        # Add placeholder email when creating the patient if it's required
+        partner_payload["email"] = f"patient_{canvas_patient.id}@canvasmedical.com"
+        partner_payload["metadata"] = self.partner_patient_metadata
+
+        base_request_url = f"{self.partner_api_base_url}/patients/v2"
+
+        # If we HAVE a patient's partner external id, we know this is an update, so we'll append it to the request URL
+        request_url = (
+            f"{base_request_url}/{system_patient_id}" if system_patient_id else base_request_url
+        )
+
+        log.info(f">>> Partner platform API request URL: {request_url}")
+        log.info(f">>> Partner platform API patient payload: json={partner_payload}")
+        log.info(
+            f">>> Partner platform API request headers: headers={self.partner_request_headers}"
+        )
+
+        resp = http.post(request_url, json=partner_payload, headers=self.partner_request_headers)
+        log.info(f">>> Partner platform API response: status_code={resp.status_code}, json={resp.json()}")
+
+        # If the request was successful, we should now have a system patient ID if we didn't before
+        if system_patient_id is None:
+            system_patient_id = resp.json().get("id")
+
+        partner_patient_exists = resp.status_code == 409
+
+        if partner_patient_exists:
+            log.info(
+                f">>> Patient already exists in {PARTNER_URL_BASE} for Canvas Patient ID {canvas_patient_id}"
+            )
+            return []
+        elif update_patient_external_identifier:
+            # queue up an effect to update the patient in canvas and add the external ID
+            external_id = CreatePatientExternalIdentifier(
+                patient_id=canvas_patient.id,
+                system=SYSTEM_URL,
+                value=str(system_patient_id)
+            )
+            return [external_id.create()]
+        else:
+            return [] # Done!
+
+
+    def sanitize_url(self, url: str) -> str:
+        """Remove a trailing slash from a URL if present."""
+        # Remove a trailing forward slash since our request paths will start with '/'
+        return url[:-1] if url[-1] == "/" else url
```

Comment:
I think we can get rid of this method and its usages. The inclusion certainly makes it more "production-grade", but at the cost of readability. Example plugins should prioritize readability, optimizing for someone to read the code and then write something very similar rather than optimizing for someone to make slight changes and ship as-is. I would argue that the MSF repo's plugins are more of that nature. These examples should answer the question "how would I...?" or "how big an effort would it be to...?", not be a drop-in solution.